libqtile/scratchpad.py

```diff
@@ -289,12 +303,23 @@ def cmd_dropdown_toggle(self, name):
         """
         Toggle visibility of named DropDown.
         """
+        if self._single:
+            for n, d in self.dropdowns.items():
+                if n != name:
+                    d.hide()
```

Comment:
This can be avoided if the dropdown specified by `name` is visible and being hidden, as no other window can be visible.

Reply:
Checking for visibility is a bit convoluted right now. I don't want to get visibility information for WindowVisibiityToggler in DropdownToggler. I don't this, this would affect any performance, as hide() will just be a no-op, if the window is already hidden.